### DISCUSSION TITLE
Registering URL encoded callback URL with Hydra client

### DISCUSSION THREAD

**Discussion by udaysarnaik** [ory/hydra#2272] [Q&A] [answered]
Hi Hydra team,

I have a question with regards to usig Hydra, please excuse me if it too trivial :slightly_smiling_face: .

I am trying to use Hydra as identity provider for my application. Plugin that i am using with my application URL encodes the redirect_uri, Thus it does not match with what i have registered with my App client in Hydra as callback URL.
I tried following option but those did not work-

1. Tried to register encoded URL as is with app client, hydra does not seem to allow that, it errors out update client call (CLI) as follows-
```
    The request failed with the following error message:
    response status code does not match any response statuses defined for this endpoint in the swagger spec (status 400): {resp:0xc000160090}
```
2. I wont be able to change the behaviour of application with sends the redirect_uri as encoded

Any pointers to handle this will be greatly appreciated!
Thank you in advance..

-Uday

**Comment by aeneasr** [comment]
Thank you for contributing to this repository by creating an issue!

Unfortunately, your issue lacks vital information, such as log files, the error message, the software version, your configuration or other pieces of the puzzle.

Please also ensure that your issue is appropriately formatted. If you do not know how to write markdown, you can find help [here](https://guides.github.com/features/mastering-markdown/).

Helping you with your problem is only possible if you share this information, and it will save a lot of time of back and forth on your as well as our end!

**Reply by udaysarnaik** [reply]
Please find error message below-

### The OAuth2 request resulted in an error.

- Error: invalid_request
- Description: The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered redirect urls.
- Hint:
- Debug:

You are seeing this page because configuration key urls.error is not set.

If you are an administrator, please read the guide to understand what you need to do. If you are a user, please contact the administrator.

**Reply by udaysarnaik** [reply]
Error message:
`time=2020-12-31T21:37:50Z level=error msg=An error occurred audience=application error=map[debug: message:invalid_request reason:The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered redirect urls. status:Bad Request status_code:400] http_request=map[headers:map[accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9 accept-encoding:gzip, deflate, br accept-language:en-GB,en;q=0.9 user-agent:Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36 x-forwarded-for:152.57.3.110 x-forwarded-proto:https] host: oryhydra method:GET path:/oauth2/auth query:Value is sensitive and has been redacted. To see the value set config key "log.leak_sensitive_values = true" or environment variable "LOG_LEAK_SENSITIVE_VALUES=true". remote:172.23.4.51:31762 scheme:https] service_name=ORY Hydra service_version=v1.9.0-alpha.3`

**Comment by udaysarnaik** [comment]
Thanks for response. The problem that i am facing is my application builds the Auth URL as-

https://hydra.example.com/oauth2/auth?client_id=quueyyyyyad8238nu8n82nxfr94&redirect_uri=https%3A%2F%2Fapplication.test.com%2Fcallback&response_type=code&scope=email+openid+profile&state=hvBvszqVR9lbFlwsD3yb02H4

Please notice encoded redirect_uri. I would not be able to change this behaviour and have non-encoded URL in here.

So this encoded URL does not exactly match with Hydra app client that i have registered (https://application.test.com/callback) with and due to this, when i try to login Hyrda throws an error that redirect_uri does not match.


As solution, i tried to register encoded URL as is with app client but hydra client update command errors out.


Any pointers would be appreciated

**Reply by aeneasr** [reply]
Please show your client - `curl https://hydra-admin.example.com/clients/quueyyyyyad8238nu8n82nxfr94`

**Comment by vinckr** [comment, marked as answer]
@udaysarnaik is this still an issue for you?

**Reply by udaysarnaik** [reply]
Hi @vinckr , the issue i was facing was resolved. I amd yet to find out the root cause for it.

**Reply by vinckr** [reply]
Ok!
Let me know if you find out more 😉
